[Doc]Update document to indicating effect of negative acking message for order sube type like exclusive, failover and key_shared. (#9681)
apache/pulsar: site2/docs/concepts-messaging.md

In the shared and Key_Shared subscription modes, you can negatively acknowledge messages individually.

> Be aware that doing negative ack on ordered subscription type like Exclusive, Failover and Key_Share can cause failed messages arrive consumer out of original order.

Reviewer: ```suggestion
Be aware that negative acknowledgment on ordered subscription types, such as Exclusive, Failover, and Key_Share, can cause failed messages to arrive consumers out of the original order.
```

Reviewer 2: `Key_Share` --> `Key_Shared`

Reviewer 2: The same name is in many other places in this PR.

Author: fixed.
